### PR TITLE
Elegant way to read output format

### PR DESCRIPTION
**Pull request by xiaoyongzhu**
Goal of this PR:
1. list all the supported cloud resources in README
2. a small refactoring to use an elegant way to read the output format

### CODE REVIEW

File: src/main/scala/com/linkedin/feathr/offline/generation/SparkIOUtils.scala [hunk 1]
```diff
@@ -19,8 +19,11 @@ object SparkIOUtils {
   }
 
   def writeDataFrame( outputDF: DataFrame, path: String, parameters: Map[String, String] = Map()): DataFrame = {
-    // if the output format is set by spark configurations, we will use that as the job output format; otherwise use avro as default for backward compatibility
-    outputDF.write.mode(SaveMode.Overwrite).format(parameters.getOrElse(OUTPUT_FORMAT, "avro")).save(path)
+
+    val output_format = outputDF.sqlContext.getConf("spark.feathr.outputFormat", "avro")
```
Comment (jaymo001) [inline src/main/scala/com/linkedin/feathr/offline/generation/SparkIOUtils.scala:23]: We should update the documentation as well?

Comment (jaymo001) [reply src/main/scala/com/linkedin/feathr/offline/generation/SparkIOUtils.scala:23]: I saw it in the other PR. NVM.

File: README.md [hunk 2]
```diff
@@ -141,6 +141,18 @@ user_item_similarity = DerivedFeature(name="user_item_similarity",
...
+
+| Feathr component             | Cloud Integrations                                                                            |
+| ---------------------------- | --------------------------------------------------------------------------------------------- |
+| Offline store – Object Store | Azure Blob Storage, Azure ADLS Gen2, AWS S3                                                   |
+| Offline store – SQL          | Azure SQL DB, Azure Synapse Dedicated SQL Pools (formerly SQL DW), Azure SQL in VM, Snowflake |
+| Online store                 | Azure Cache for Redis                                                                         |
+| Feature Registry             | Azure Purview                                                                                 |
+| Compute Engine               | Azure Synapse Spark Pools, Databricks                                                         |
+| Machine Learning Platform    | Azure Machine Learning, Jupyter Notebook                                                      |
+| File Format                  | Parquet, ORC, Avro, Delta Lake                                                                |
```
Comment (hangfei) [inline README.md:154]: this is a bit hard to make sense. probably defer this to a diff rb?

Comment (xiaoyongzhu) [reply README.md:154]: Ya I think it's like recording all the supported output format in the README file as @jaymo001 suggested, and we don't have that table for that hence also adding that table in the README file.

**Review by hangfei — APPROVED** [review]

**Pull request merged**